ivy/functional/ivy/creation.py

```diff
@@ -1691,13 +1691,53 @@ def full(
     )
 
 
+@handle_exceptions
 @handle_backend_invalid
 @handle_nestable
-@handle_array_like_without_promotion
 @handle_out_argument
 @to_native_arrays_and_back
 @handle_array_function
 @handle_device_shifting
+def to_dlpack(
+    x: Union[ivy.Array, ivy.NativeArray], /, *, out: Optional[ivy.Array] = None
+):
+    """
+    Return PyCapsule Object.
+
+    Parameters
+    ----------
+    x  object
+        input (array) object.
+    out
+        optional output array, for writing the result to. It must have a shape that the
+        inputs broadcast to.
+
+    Returns
+    -------
+    ret
+        Return PyCapsule Object.
+
+        .. admonition:: Note
+           :class: note
+
+           The returned array may be either a copy or a view. See
+           :ref:`data-interchange` for details.
+
+
+    This function conforms to the `Array API Standard
+    <https://data-apis.org/array-api/latest/>`_. This docstring is an extension of the
+    `docstring <https://data-apis.org/array-api/latest/
+    API_specification/generated/array_api.from_dlpack.html>`_
+    in the standard.
+
+    Both the description and the type hints above assumes an array input for simplicity,
+    but this function is *nestable*, and therefore also accepts :class:`ivy.Container`
+    instances in place of any of the arguments.
+    """
+    return current_backend(x).to_dlpack(x, out=out)
+
+
+@handle_backend_invalid
```

Comment:
we should also add all applicable decorators to `from_dlpack` if it accepts an `ivy.Array`

Reply:
these decorators operate on array but on `PyCapsule` object as it's not array most decorators throw errors.